### DISCUSSION TITLE
Tools synergy, or how to get inspiration from others

### DISCUSSION THREAD

**Discussion by ccoVeille** [jargonLint/jargonLint#1] [Ideas]
Hi everyone

Sorry for bringing you into my idea, but I think we could have interest in working together.

I have a huge interest in document linting, and I like your tools

I tried multiple tool:
- [textlint-rule-terminology](https://github.com/sapegin/textlint-rule-terminology) (hi @sapegin :wave:)
- [vale](https://github.com/errata-ai/vale)  (hi @jdkato :wave: )
- [autocorrect](https://github.com/huacnlee/autocorrect) (hi @huacnlee :wave: )
   even if autocorrect is intended to be used with CJK (aka "Asian characters"), it provides the same feature for fixing brand cases.
- [languagetool](https://github.com/languagetool-org/languagetool) (hi @jaumeortola :wave:)
- [cspell](https://github.com/streetsidesoftware/cspell) (hi @arkid15r and @Jason3S)
- [typos](https://github.com/crate-ci/typos) (hi @epage :wave:)
- [codespell](https://github.com/codespell-project/) (hi @luzpaz :wave:)
- [misspell](https://github.com/client9/misspell/) (hi @silverwind :wave: I know you have interest for this)
- [suze-vale tbazant's branch](https://github.com/openSUSE/suse-vale-styleguide/tree/tbazant-test-spellchecking) (hi @tbazant :wave:) because his ideas are close to what I want to achieve when I started thinking about this project

I use your tools often because I love fixing things in public repositories, but also on my own codebase.

You may not know each other tools, but you guy rocks.

I created jargon (this project) because I plan to work on way to maintain a list of brands and common fixes that would make sense to provide to anyone interested in working on such fix.

Which mean I don't plan to create a tool like yours, but to provide data that could be used by tools like yours.

I would use your tools as source of information, because they have valuable examples.

- textlint-rule-terminology: https://github.com/sapegin/textlint-rule-terminology/blob/master/terms.jsonc
- autocorrect: https://github.com/huacnlee/autocorrect/blob/main/.autocorrectrc.template
- cspell-dicts: https://github.com/streetsidesoftware/cspell-dicts/blob/main/dictionaries/software-terms/src/computing-acronyms.txt
- suse-vale: https://github.com/openSUSE/suse-vale-styleguide/blob/tbazant-test-spellchecking/common%2FTerms.yml
- languagetool:

   https://community.languagetool.org/rule/show/GIT_HUB?lang=en&subId=1
   https://community.languagetool.org/rule/show/IN_FACEBOOK?lang=en-US&subId=140
   https://community.languagetool.org/rule/show/GIT_HUB?lang=en&subId=1
   https://community.languagetool.org/rule/show/GITHUB?lang=en&subId=1
   https://community.languagetool.org/rule/show/MARKDOWN_NNP?lang=en&subId=2
   https://github.com/languagetool-org/languagetool/blob/master/languagetool-language-modules/en/src/main/resources/org/languagetool/rules/en/grammar.xml
   https://github.com/languagetool-org/languagetool/blob/master/languagetool-language-modules/en/src/main/resources/org/languagetool/resource/en/specific_case.txt

But also plan to use Wikidata and maybe use Docker API to extract the name of brands with their exact cases

I might use also awesome lists like
- https://github.com/paradite/frontend-encyclopedia
- https://github.com/igorbarinov/awesome-data-engineering

My idea with my tool was initially to build a [vale package](https://vale.sh/docs/topics/packages/), but as I continue thinking about the tool, I'm thinking about maybe working on a database and then provide something that could be copied and pasted to vale, textlint-rule-terminology, autocorrect.

Then people, and you guys might import the ones you think interesting for your projects.

I'm curious if you guys are interested in such "database" / "resources", that's why I created https://github.com/ccoVeille/jargon

My idea is also to maybe allow you to bring everyone with me so when something is added to someone project, I will replicate it in jargon, but maybe later, you guys may open PR in jargon (I'm dreaming maybe)

So I'm somehow inviting you guys to talk about it, here.

I'm unsure where things will bring me, but I wanted to give a start to this project that I keep thinking about for months now.

Why jargon ? Because it's an English and a French word (I'm French) and I like how old-fashioned it sounds. (I'm open for suggestion see #2)

I created a repository, but an organization might make more sense at some point, we will figure it out together if you guys find interest in it

Thanks

Feel free to invite anyone you may think who might have interest in this idea.

Christophe Colombier

**Comment by ccoVeille** [comment]
Please bring anyone interested in this project by pinging them in this thread

**Reply by ccoVeille** [reply]
adding @silverwind I failed to include you earlier because of a typo in your name.

**Reply by ccoVeille** [reply]
Adding @tbazant because his ideas are close to what I want to achieve

https://github.com/openSUSE/suse-vale-styleguide/tree/tbazant-test-spellchecking

**Comment by silverwind** [comment]
I'm not sure I will be of much help. I have been using misspell (https://github.com/golangci/misspell is the current fork) mostly as a consumer. What I value in it is that its dictionary has a near-zero amount of false-positives contrary to a number of other tools in this list which often lean on the side of reporting more than less, requiring tedious exclusions to be maintained which is not a scalable approach.

So if a new tool is made, I would definitely welcome if it had a `absolutely-no-false-positives` category in the fashion of misspell's list.

**Reply by ccoVeille** [reply]
Thanks for your feedback. I understand your point of view

For record, I also have the project to maintain misspell by updating the code used to generate the list of words from the exact way client9 did: Wikipedia misspelling and co.

But it would be out of "jargon" I think. I will ping you when I will work on this

**Reply by ccoVeille** [reply]
> So if a new tool is made, I would definitely welcome if it had a absolutely-no-false-positives category in the fashion of misspell's list.

Did you try typos?

https://github.com/crate-ci/typos

I'm very happy about it and its approach, that match what you are describing and looking for

**Comment by ccoVeille** [comment]
Hi @alexvonme

You might find interest in my future project

**Reply by alexvonme** [reply]
I do! In fact, this was the project I was mentioning earlier while implying that we both want to reduce needless re-iterations.

**Reply by ccoVeille** [reply]
I provided you information about how to reach me on the slack discussion we had earlier on Gopher's Slack
